toolchain/base/timings.h

```diff
@@ -14,24 +16,24 @@ namespace Carbon {
 // Helps track timings for a compile.
 class Timings {
  public:
-  // Records a timing for a scope, if the timings object is valid.
+  // Records a timing for a scope, if the timings object is present.
   class ScopedTiming {
    public:
-    explicit ScopedTiming(std::optional<Timings>* timings,
-                          llvm::StringRef label)
+    // The `timings` may be null, in which case the `ScopedTiming` is a no-op.
+    explicit ScopedTiming(Timings* timings, llvm::StringRef label)
         : timings(timings),
           label(label),
-          start(*timings ? std::chrono::steady_clock::now()
-                         : std::chrono::steady_clock::time_point::min()) {}
+          start(timings ? std::chrono::steady_clock::now()
+                        : std::chrono::steady_clock::time_point::min()) {}
 
     ~ScopedTiming() {
-      if (*timings) {
-        (*timings)->Add(label, std::chrono::steady_clock::now() - start);
+      if (timings) {
+        timings->Add(label, std::chrono::steady_clock::now() - start);
       }
     }
 
    private:
-    std::optional<Timings>* timings;
+    Timings* timings;
     llvm::StringRef label;
     std::chrono::steady_clock::time_point start;
```

Comment:
Small thing while you're in here, I should've had these class members named with underscores (`timings_` etc). Would you be okay fixing that as long as you're refactoring here? (if you'd rather not, feel free to merge without it)

Reply:
Oh yeah, done.